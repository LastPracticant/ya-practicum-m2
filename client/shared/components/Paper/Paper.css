.paper {
    border-radius: var(--space-xs);
    padding: var(--space-unit);

    &_small {
        width: 340px;
        padding: var(--space-xs);
    }

    &_theme_dark {
        background-color: #444444d4;
        color: #fff;
        box-shadow: 0 2px 7px #000;
    }
}
