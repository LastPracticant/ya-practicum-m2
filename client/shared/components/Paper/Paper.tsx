import React, { FC, memo } from 'react'
import { createUseStyles } from 'react-jss'
import Theme from 'SharedStyles/theme/Theme'

const useStyles = createUseStyles({
    'YlpPaper-root': {
        background: Theme.backgroundColor,
        borderRadius: Theme.radius,
        boxShadow: Theme.shadow
    }
})

export const Paper: FC<{ style?: React.CSSProperties, className?: string }> = memo(
    ({ children, style, className }) => {
        const classes = useStyles()
        return (
            <div className={[classes['YlpPaper-root'], className].join(' ')} style={style}>
                {children}
            </div>
        )
    }
)
