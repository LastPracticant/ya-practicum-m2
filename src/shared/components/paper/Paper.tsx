import React, { FC, memo } from 'react'
import Theme from '../../styles/theme/Theme'

export const Paper: FC<{ style: React.CSSProperties }> = memo(
  ({ children, style }) => {
    return (
      <div
        style={{
          background: Theme.backgroundColor,
          borderRadius: Theme.radius,
          boxShadow: Theme.shadow,
          ...style
        }}
      >
        {children}
      </div>
    )
  }
)
