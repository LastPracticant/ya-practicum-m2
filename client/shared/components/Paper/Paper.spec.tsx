import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Paper } from './Paper';

afterEach(cleanup);

it('Render Paper', () => {
    const { queryByLabelText, getByLabelText } = render(
        <Paper title="Test Paper" sizes="full">
            <span>LP</span>
        </Paper>,
    );
    console.log(queryByLabelText, getByLabelText);
});
