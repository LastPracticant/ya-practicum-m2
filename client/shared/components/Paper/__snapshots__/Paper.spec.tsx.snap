// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <h2
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    Test Paper
  </h2>
  <section
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <span>
      LP
    </span>
  </section>
</div>
`;
